src/mimir_config.py

```diff
@@ -108,6 +108,14 @@ class _S3ConfigData(BaseModel):
     secret_access_key: str = Field(alias="secret-key")
     bucket_name: str = Field(alias="bucket")
     region: str = ""
+    insecure: str = "false"
+
+    @model_validator(mode="before")  # pyright: ignore
+    @classmethod
+    def set_insecure(cls, data: Any) -> Any:
+        if isinstance(data, dict) and data.get("endpoint", None):
+            data["insecure"] = "true" if data["endpoint"].startswith("http://") else "false"
+        return data
```

Comment:
- Not sure how a `model_validator` works, but do we need to `.copy()` the mutable first?
- Why do we need the instance check i.e why `data` is of type `Any` and not `dict`?

Reply:
> * Not sure how a `model_validator` works, but do we need to `.copy()` the mutable first?
> 
> * Why do we need the instance check i.e why `data` is of type `Any` and not `dict`?

I think [this section of the documentation](https://docs.pydantic.dev/latest/concepts/validators/#model-validators) answer both questions:

*"Before model validators are passed the raw input which is often a dict[str, Any] but could also be an instance of the model itself (e.g. if UserModel.model_validate(UserModel.construct(...)) is called) or anything else since you can pass arbitrary objects into model_validate."*

Reply:
I would rather flip this to
```python
data["insecure"] = "false" if data["endpoint"].startswith("https://") else "true"
```
just in case the **endpoint** comes is in without any schema (not sure this can happen, but maybe it's a bit more robust? what do you think?)